File: frontend/classes/Senator.ts
export interface SenatorData {
  id: number
  game: number
  name: string
  code: string
  faction: number | null
  alive: boolean
  military: number
  oratory: number
  loyalty: number
  influence: number
  popularity: number
  knights: number
  talents: number
  votes: number
  status_items: string[]
  titles: string[]
  location: string
  display_name: string
}

class Senator {
  id: number
  game: number
  name: string
  code: string
  faction: number | null
  alive: boolean
  military: number
  oratory: number
  loyalty: number
  influence: number
  popularity: number
  knights: number
  talents: number
  votes: number
  statusItems: string[]
  titles: string[]
  location: string
  displayName: string

  constructor(data: SenatorData) {
    this.id = data.id
    this.game = data.game
    this.name = data.name
    this.code = data.code
    this.faction = data.faction
    this.alive = data.alive
    this.military = data.military
    this.oratory = data.oratory
    this.loyalty = data.loyalty
    this.influence = data.influence
    this.popularity = data.popularity
    this.knights = data.knights
    this.talents = data.talents
    this.votes = data.votes
    this.statusItems = data.status_items
    this.titles = data.titles
    this.location = data.location
    this.displayName = data.display_name
  }
}

export default Senator

File: backend/rorapp/serializers/senator.py
from rest_framework import serializers

from rorapp.models import Senator


class SenatorSerializer(serializers.ModelSerializer):
    class Meta:
        model = Senator
        fields = [
            "id",
            "game",
            "family_name",
            "statesman_name",
            "family",
            "code",
            "faction",
            "alive",
            "military",
            "oratory",
            "loyalty",
            "influence",
            "popularity",
            "knights",
            "talents",
            "votes",
            "status_items",
            "titles",
            "concessions",
            "corrupt_concessions",
            "location",
            "display_name",
        ]
